examples/avr/can/mcp2515/main.cpp

```diff
@@ -32,6 +32,20 @@ FLASH_STORAGE(uint8_t canFilter[]) =
 	MCP2515_FILTER_EXTENDED(0),	// Mask 1
 };
 
+xpcc::Abandonment can_assertion_handler(const char *module, const char *location, const char *failure, uintptr_t)
+{
+    if (!strncmp(module,   "mcp2515",  7) and
+    	!strncmp(location, "init",     4) and
+    	!strncmp(failure,  "readback", 8))
+    {
+        return xpcc::Abandonment::Fail;
+    }
+    return xpcc::Abandonment::DontCare;
+}
+
+// Register assertion handler with system
+XPCC_ASSERTION_HANDLER(can_assertion_handler);
```

Comment:
The `xpcc_assert` by default behaves like a normal `assert`, it abandons program execution if it fails. You don't need to do anything special to enable this behavior. 

Only if you want to ignore an assertion, then you can define an assertion handler that returns `xpcc::Abandonment::Ignore` when it matches the assertion identifiers.